src/main/kotlin/de/cyface/collector/verticle/CollectorApiVerticle.kt

```diff
@@ -232,24 +325,82 @@ class CollectorApiVerticle(
+        router.post(MEASUREMENTS_ENDPOINT)
+            .consumes("application/json; charset=UTF-8")
+            .handler(LoggerHandler.create())
+            // Read request body only once and before async calls or pause/resume must be used see [DAT-749]
+            .handler(preRequestBodyHandler)
+            .handler(oauth2Handler)
+            .handler(authorizationHandler)
+            .handler(PreRequestHandler(storageService, config.measurementPayloadLimit))
+            .failureHandler(failureHandler)
+    }
+
+    /**
+     * Adds a handler for an endpoint and makes sure that handler is wrapped in the correct authentication handlers.
+     *
+     * @param router The `Router` to register the handler to.
+     * @param oauth2Handler The handler to authenticate the user using an OAuth2 endpoint.
+     * @param authorizationHandler The handler used to authorize a user after successful authentication.
+     * @param failureHandler The handler to add to handle failures.
+     * @param storageService Service used to write the received data.
+     */
+    private fun registerMeasurementHandler(
+        router: Router,
+        oauth2Handler: OAuth2AuthHandler,
+        authorizationHandler: AuthorizationHandler,
+        failureHandler: ErrorHandler,
+        storageService: DataStorageService
+    ) {
+        // The path pattern ../(sid)/.. was chosen because of the documentation of Vert.X SessionHandler
+        // https://vertx.io/docs/vertx-web/java/#_handling_sessions
+        val measurementHandler = config.measurementPayloadLimit
+        router.putWithRegex(String.format(Locale.ENGLISH, "\\%s\\/\\([a-z0-9]{32}\\)\\/", MEASUREMENTS_ENDPOINT))
+            .consumes("application/octet-stream")
+            // Not using BodyHandler as the `request.body()` can only be read once and the {@code #handler} does so.
+            .handler(LoggerHandler.create())
+            .handler(oauth2Handler)
+            .handler(authorizationHandler)
+            .handler(MeasurementHandler(storageService, measurementHandler))
+            .handler(StatusHandler(storageService))
+            .failureHandler(failureHandler)
+    }
+
+    /**
+     * Adds a handler for an endpoint and makes sure that handler is wrapped in the correct authentication handlers.
+     *
+     * @param router The `Router` to register the handler to.
+     * @param jwtAuthHandler The handler to authenticate the user using JWT tokens.
+     * @param authorizationHandler The handler used to authorize a user after successful authentication.
+     * @param failureHandler The handler to add to handle failures.
+     * @param storageService Service used to write the received data.
+     */
+    private fun registerPreRequestHandlerV3(
+        router: Router,
+        jwtAuthHandler: JWTAuthHandler,
+        authorizationHandler: de.cyface.collector.v6.handler.AuthorizationHandler,
+        failureHandler: ErrorHandler,
+        storageService: DataStorageService
```

Comment:
can be ignored, didn't change (just added a copy for v3 as I changed it for the current API `v4`.